example-2/evm/src/contracts/lendingHub/Hub.sol

```diff
@@ -199,7 +225,11 @@ contract Hub is HubStructs, HubMessages, HubSetters, HubGetters {
         VaultAmount memory globalAmounts = getGlobalAmounts(assetAddress);
         globalAmounts.borrowed -= normalizedAmount;
 
-        // TODO: token transfers (do you need this here? you should probably just transfer tokens directly to the lending protocol via relayer)
+
+    }
+
+    function transferTokens(address receiver, address assetAddress, uint256 amount) internal {
```

Comment:
on the redemption of tokens on the target chains, two options: 1. you could send P3 to target chain spoke contract to handle repayment; 2. you could send directly to recipient on target chain via token bridge transfer.